### PR TITLE
Export transaction to file [ITERATION 1]

### PR DESCRIPTION
**Pull request by sarveshwar-singh**
Reference:
doc: https://github.com/Emurgo/yoroi-frontend/pull/189

API Implementation: https://github.com/Emurgo/yoroi-frontend/pull/210 + https://github.com/Emurgo/yoroi-frontend/tree/feature/tx-export

### CODE REVIEW

**Comment by sarveshwar-singh** [comment]
@vantuz-subhuman @nicarq 

If there any reason for not using `// @flow` flow target in the following files?

https://github.com/Emurgo/yoroi-frontend/pull/222/files#diff-a2e926e9dcc51b770ec51e30643a9a5b
https://github.com/Emurgo/yoroi-frontend/pull/222/files#diff-b45881d223bc76e38c5909a06d0cf197
https://github.com/Emurgo/yoroi-frontend/pull/222/files#diff-162ee01e35d64f85ecd5a1753a69cc09
https://github.com/Emurgo/yoroi-frontend/pull/222/files#diff-0889d76a390387291d112fe516e3253b
https://github.com/Emurgo/yoroi-frontend/pull/222/files#diff-1ffd4de59e510236411092ad907a5b1f

**Comment by nicarq** [comment]
@sarveshwar-singh really good catch!!!! yeah, I think we should. cc @vantuz-subhuman

File: app/api/export/index.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+   * Request object MUST contains:
+   * - rows: array of export-data
+   * - fileName: only name without extension
+   *
+   * No result will be returned. File is sent to user as side-effect.
+   */
+  exportTransactions = async (
+    request : ExportTransactionsRequest
+  ): Promise<ExportTransactionsResponse> => {
+    const { rows, format, fileType, fileName } = request;
```
Comment (sarveshwar-singh) [inline app/api/export/index.js]: @vantuz-subhuman @nicarq 

What is the reason behind passing `fileName` from outside?

User don't have control over file name selection.
Even if `exportTransactions` API accepts `fileName` then it should be optional.

So thinking to make `fileName` optional and add `fileName` creation logic at API side.

File: app/api/export/index.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// @flow
+import '../ada/lib/test-config'; // TODO: needs to import this from ADA API, should be somewhere common at TOP level
```
Comment (emurgo-web) [inline app/api/export/index.test.js]: what is the plan for the TODOs in the PR? are being tracked somehow?

File: app/api/export/index.test.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  });
+  expect(await extractStringFromBlob(data)).toEqual('"qwe","rty","qaz"\n"a","b","c"\n"d","e","f"');
+  expect(fileType).toEqual('csv');
+});
+
+async function extractStringFromBlob(b): Promise<string> {
+  return new Promise(resolve => {
+    const reader = new FileReader();
+    reader.addEventListener('loadend', () => {
+      // TODO
```
Comment (nicarq) [inline app/api/export/index.test.js]: TODO + Commented code.

File: app/api/export/index.test.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      // resolve(String.fromCharCode.apply(null, new Uint8Array(reader.result)));
+      if (typeof reader.result === 'string') {
+        resolve(reader.result);
+      }
+    });
+    // TODO
```
Comment (nicarq) [inline app/api/export/index.test.js]: TODO + Commented code.

File: app/api/export/index.test.js (hunk 2 above)
Comment (sarveshwar-singh) [reply app/api/export/index.test.js]: was not tracked outside.
Now added : https://github.com/Emurgo/yoroi-frontend/issues/250

File: app/api/export/index.test.js (hunk 4 above)
Comment (sarveshwar-singh) [reply app/api/export/index.test.js]: just kept for @vantuz-subhuman review.
can delete now.

File: app/api/export/index.test.js (hunk 3 above)
Comment (sarveshwar-singh) [reply app/api/export/index.test.js]: just kept for @vantuz-subhuman review.
can delete now.

**Pull request merged**